api/base/serializers.py

```diff
@@ -88,7 +105,35 @@ def to_internal_value(self, data):
         return super(JSONAPIListField, self).to_internal_value(data)
 
 
-class JSONAPIHyperlinkedIdentityField(ser.HyperlinkedIdentityField):
+class AuthorizedCharField(ser.CharField):
+    """
+    Passes auth of the logged-in user to the object's method
+    defined as the field source.
+
+    Example:
+        content = AuthorizedCharField(source='return_content')
+    """
+
+    def __init__(self, **kwargs):
+        super(AuthorizedCharField, self).__init__(**kwargs)
+
+    def get_attribute(self, obj):
+        user = self.context['request'].user
+        auth = Auth(user)
+        field_source_method = getattr(obj, self.source)
```

Comment:
Why does this use getattr? Since default is None, line 124 would fail if no `self.source` value was set; this attribute does not appear to be optional.